kensu/utils/kensu.py

```diff
@@ -285,8 +289,22 @@ def init_context(self, process_name=None, user_name=None, code_location=None, ge
             self.report_process_info()
 
     def report_process_info(self):
-        for e in [self.user, self.code_base, self.code_version, self.process] + self.projects + [self.process_run]:
-            e._report()
+        b = BatchEntityReport()
+        if self.code_base:
+            b.code_bases = [BatchCodeBase(self.timestamp, self.code_base)]
+        if self.code_version:
+            b.code_versions = [BatchCodeVersion(self.timestamp, self.code_version)]
+        if self.user:
+            b.users = [BatchUser(self.timestamp, self.user)]
+        projects = [p for p in self.projects if p]
+        if projects:
+            b.projects = [BatchProject(self.timestamp, p) for p in projects]
+        if self.process:
+            b.processes = [BatchProcess(self.timestamp, self.process)]
+        if self.process_run:
+            b.process_runs = [BatchProcessRun(self.timestamp, self.process_run)]
```

Comment:
we could later add some pythonic helpers to build batch like `b.add(entity)`, so basically the code could be like it was before above